server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java

```diff
@@ -363,9 +373,38 @@ private SearchWsResponse doHandle(SearchRequest request, Request wsRequest) {
     // FIXME allow long in Paging
     Paging paging = forPageIndex(options.getPage()).withPageSize(options.getLimit()).andTotal((int) result.getHits().getTotalHits());
 
+    setRuleKeyForRuleFacets(facets);
+
     return searchResponseFormat.formatSearch(additionalFields, data, paging, facets);
   }
 
+  private void setRuleKeyForRuleFacets(Facets facets) {
+    // The facet for PARAM_RULES contains the id of the rule as the key
+    // We need to update the key to be a RuleKey
+    LinkedHashMap<String, Long> rulesFacet = facets == null ? null : facets.get(PARAM_RULES);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Change this condition so that it does not always evaluate to "false" ([squid:S2583](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2583))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Feh%2FSONAR-10303&issues=AWEx4FzDr8h_fVHZ4pDT&open=AWEx4FzDr8h_fVHZ4pDT)

[ISSUE_KEY]: AWEx4FzDr8h_fVHZ4pDT